src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs

```diff
@@ -159,6 +159,16 @@ private static IEnumerable<Type> FindConfigurationServices(Type type)
                     }
                 }
             }
+
+            // Extracted the suppression to a local function as trimmer currently doesn't handle suppressions
+            // on iterator methods correctly.
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+                Justification="This method only looks for interfaces referenced in its code. " +
+                    "The trimmer will keep the interface and thus all of its implementations in that case. " +
+                    "The call to GetInterfaces may return less results in trimmed apps, but it will " +
+                    "include the interfaces this method looks for if they should be there.")]
+            static Type[] GetInterfacesOnType (Type t)
```

Comment:
```suggestion
            static Type[] GetInterfacesOnType(Type t)
```

Reply:
We need to make this part of the process after working in the linker repo:

<img src="https://blog.gosafe.com/hs-fs/hubfs/Campaign_1_Facility_Safety/C1_Blog%20Images/Eyewash%20Gif%203.gif?width=1024&name=Eyewash%20Gif%203.gif">